### PR TITLE
Dropped unnecessary editor change event for autocomplete

### LINKED ISSUE #2194: Removing autocomplete inserted match characters with enabled throttling throws an error
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Use build editor
2. Open test: http://tests.ckeditor.test:1410/tests/plugins/autocomplete/manual/throttle
3. Open console
4. Insert autocomplete name
5. Wait moment and start to remove characters with backspace

### Expected result

no errors in console

### Actual result

there appears errors in console;

> Uncaught DOMException: Failed to execute 'setEnd' on 'Range': The offset 2 is larger than the node's length (1).
>     at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.getClientRects (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:187:3)
>     at getViewPosition (http://tests.ckeditor.test:1410/apps/ckeditor/plugins/autocomplete/plugin.js?t=I629:14:470)
>     at g.updatePosition (http://tests.ckeditor.test:1410/apps/ckeditor/plugins/autocomplete/plugin.js?t=I629:18:76)
>     at f.viewRepositionListener (http://tests.ckeditor.test:1410/apps/ckeditor/plugins/autocomplete/plugin.js?t=I629:11:217)
>     at f.<anonymous> (http://tests.ckeditor.test:1410/apps/ckeditor/plugins/autocomplete/plugin.js?t=I629:9:107)
>     at a.h (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:10:70)
>     at a.<anonymous> (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:11:433)
>     at a.window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.editor.CKEDITOR.editor.fire (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:13:97)
>     at CKEDITOR.plugins.undo.UndoManager.type (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:1195:102)
>     at CKEDITOR.plugins.undo.NativeEditingHandler.onKeyup (http://tests.ckeditor.test:1410/apps/ckeditor/ckeditor.js:1205:439)
> 

## Other details

* Browser: Opera
* OS: MacOS
* CKEditor version: 4.10.0
* Installed CKEditor plugins: …

### LINKED ISSUE #2256: Got "Uncaught DOMException: Failed to execute 'setEnd' on 'Range'" when operating on a mention
## Type of report

Bug

## Provide detailed reproduction steps (if any)

1. Open https://sdk.ckeditor.com/samples/mentions.html.
2. Make sure that the developer console is open.
3. Mention one of the users, say, "@ ajames" (without the space).
4. Delete the mention added in step 3 one character at a time. 

May need to repeat steps 3 and 4 multiple times for the Exception to happen.

### Expected result

No errors in developer console.

### Actual result

Got the following errors:

Uncaught DOMException: Failed to execute 'setEnd' on 'Range': The offset 6 is larger than the node's length (5).
    at window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.dom.range.getClientRects (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:187:3)
    at getViewPosition (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/plugins/autocomplete/plugin.js?t=I6D6:14:470)
    at g.updatePosition (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/plugins/autocomplete/plugin.js?t=I6D6:18:76)
    at f.viewRepositionListener (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/plugins/autocomplete/plugin.js?t=I6D6:11:217)
    at f.<anonymous> (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/plugins/autocomplete/plugin.js?t=I6D6:9:107)
    at a.h (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:10:70)
    at a.<anonymous> (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:11:433)
    at a.window.CKEDITOR.window.CKEDITOR.dom.CKEDITOR.editor.CKEDITOR.editor.fire (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:13:97)
    at CKEDITOR.plugins.undo.UndoManager.type (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:1079:373)
    at CKEDITOR.plugins.undo.NativeEditingHandler.onKeyup (https://sdk.ckeditor.com/vendor/ckeditor/4.10.0/ckeditor.js:1090:121)

## Other details

* Browser: Chrome, Safari

### LINKED ISSUE #2474: Autocomplete shouldn't listen on change event
## Type of report

Task

## Provide description of the task

There is no longer reason to reposition autocomplete view on `change` event. Autocomplete dropdown is pinned to the beginning of the match #2028 so we can safely remove this listener from the editor.

## Other details

* CKEditor version: 4.11.0
* Installed CKEditor plugins: autocomplete

### PR DESCRIPTION
**Pull request by jacekbogdanski**
## What is the purpose of this pull request?

Task

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_contributing_code.html#tests),
[how to set the testing environment](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html) and
[how to create tests](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_tests.html#creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [ ] Manual tests

## What changes did you make?

There is no need for updating the view position on `change` event which was called out of sync with text matching.

Closes #2474

### CODE REVIEW

**Comment by jacekbogdanski** [comment]
Good catch, I also rebased branch into upstream.

**Comment by Comandeer** [comment]
Rebased onto latest `major`.

File: tests/plugins/autocomplete/manual/removematch.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+@bender-tags: 4.10.2, bug, 2194
+@bender-ui: collapsed
+@bender-ckeditor-plugins: wysiwygarea, toolbar, basicstyles, autocomplete, textmatch
+@bender-include: _helpers/utils.js
+
+1. Open the console.
+1. Focus the editor.
+1. Type `@`.
+1. Accept `@anna` dropdown option.
+1. Start deleting inserted match using `backspace` key.
```
Comment (Comandeer) [inline tests/plugins/autocomplete/manual/removematch.md]: The important bit of information is missing: steps should be repeated several times (3-5), as error occurs randomly.

**Comment by jacekbogdanski** [comment]
Actually, I incorrectly refactored listeners for view reposition on scroll. I restored it with some comment to be more carefully next time.

**Comment by engineering-this** [comment]
I reopened issue ticket as I can still reproduce it. cc @jacekbogdanski @Comandeer

**Comment by pupi1985** [comment]
Hi. I was wondering if there was any estimated milestone for this to be merged/fixed. Maybe 4.11.2? Or 4.12.0? Any estimation will be appreciated. BTW, I won't be expecting you to commit on a date, I'd just find that useful to plan my own releases. Thank you!

**Comment by f1ames** [comment]
Rebased and retargeted onto latest `master`.

**Comment by jacekbogdanski** [comment]
AFAIK this issue is no longer responsible for fixing the issue but code cleanup :thinking: The real issue with error is caused by https://github.com/ckeditor/ckeditor4/issues/2194 and should be resolved by another PR to autocomplete throttling logic.

I propose to still treat it as a cleanup task.

**Comment by f1ames** [comment]
Rebased and retargeted onto latest `major`.

**Comment by JDanP** [comment]
Hi @msamsel ! Im having this issue and I can realize the fix was made but on 10 Jan a revert was applied. Can you tell why? It seems that really fix the problem. Really need help here.

**Pull request merged**